src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp

```diff
@@ -240,22 +247,18 @@ void QgsExternalResourceConfigDlg::setConfig( const QVariantMap &config )
   if ( config.contains( QStringLiteral( "RelativeStorage" ) ) )
   {
     const int relative = config.value( QStringLiteral( "RelativeStorage" ) ).toInt();
-    if ( ( QgsFileWidget::RelativeStorage )relative == QgsFileWidget::Absolute )
-    {
-      mRelativeGroupBox->setChecked( false );
-    }
-    else
-    {
-      mRelativeGroupBox->setChecked( true );
-      mRelativeButtonGroup->button( relative )->setChecked( true );
-    }
+
+    if ( ( QgsFileWidget::RelativeStorage )relative == QgsFileWidget::RelativeDefaultPath )
+      enableCbxItem( mStoragePathCbx, 2, true );
```

Comment:
Is this really necessary to enable it here? It should be done when `mRootPath` or `mRootPathExpression` is updated no?

Reply:
I think it should. I wanted to be safe (or actually I was unsure of what was happening).
Looking to the code, I can't find what is stored in config when people use mRootPathExpression instead of mRootPath. Am I missing something?

Reply:
`mRootPathExpression` is related to the data defined property button, it's stored under the data defined property configuration and override the classic configuration mRootPath. 

To sum up, mRootPath and mRootPathExpression target the same config, if one is not null, the relativeDefaultPath should be enabled. and so the signal connection should deal with this.

Reply:
Yes, this is what i noticed while testing